h2o-algos/src/main/java/hex/tree/SharedTree.java

```diff
@@ -409,10 +409,10 @@ protected final void resumeFromCheckpoint(SharedTree st) {
     protected final void scoreAndBuildTrees(boolean oob) {
       for( int tid=0; tid< _ntrees; tid++) {
         // During first iteration model contains 0 trees, then 1-tree, ...
-        boolean scored = doScoringAndSaveModel(false, oob, _parms._build_tree_one_node);
+        final boolean scored = doScoringAndSaveModel(false, oob, _parms._build_tree_one_node);
         if (scored && ScoreKeeper.stopEarly(_model._output.scoreKeepers(), _parms._stopping_rounds, getProblemType(), _parms._stopping_metric, _parms._stopping_tolerance, "model's last", true)) {
-          doScoringAndSaveModel(true, oob, _parms._build_tree_one_node);
```

Comment:
Is the parameter `finalScoring` still useful after your changes?

Reply:
Yes, it forces the scoring to run regardless of other criteria. Still needed.